crates/downloader/src/block_buffer.rs

```diff
@@ -0,0 +1,77 @@
+use std::collections::BTreeMap;
+
+use fuel_core_compression::{
+    VersionedBlockPayload,
+    VersionedCompressedBlock,
+};
+use fuel_core_types::fuel_types::BlockHeight;
+
+#[derive(Default)]
+pub struct BlockBuffer {
+    blocks: BTreeMap<BlockHeight, VersionedCompressedBlock>,
+}
+
+impl BlockBuffer {
+    pub fn push(&mut self, block: VersionedCompressedBlock) {
+        self.blocks.insert(*block.height(), block);
+    }
+
+    /// Given "high water mark" block height, remove all blocks below it.
+    pub fn prune_below(&mut self, height: BlockHeight) {
+        self.blocks = self.blocks.split_off(&height);
+    }
+
+    /// If the first entry has given height, remove and return it.
+    pub fn pop_height(
+        &mut self,
+        height: BlockHeight,
+    ) -> Option<VersionedCompressedBlock> {
+        if let Some(head) = self.blocks.first_entry() {
+            if head.key() == &height {
+                Some(head.remove())
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use fuel_core_compression::{
+        // CompressedBlockPayloadV0,
```

Comment:
```suggestion
```

Reply:
addressed in [a165fa9](https://github.com/FuelLabs/network-watchtower/pull/10/commits/a165fa92704c4f1cea7f5de1a3183a715c5e59d9)